lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -2584,6 +2584,93 @@ struct ConvertReductionOp : ConversionPattern {
 };
 } // namespace
 
+// Computes maxpool2d for AtenMaxPool2dOp and AtenMaxPool2dWithIndicesOp.
+template <typename OpTy>
+static LogicalResult
+computeMaxPool2d(OpTy op, Location loc, ConversionPatternRewriter &rewriter,
+                 Value self, Value ceilMode, Value &result) {
+  Type elementType = self.getType().cast<RankedTensorType>().getElementType();
+  if (!elementType.isa<mlir::FloatType>())
+    return op.emitError("unimplemented: non-floating point type");
+
+  // Pattern match against the op's original operands, because otherwise we
+  // will get the lowered version of the operands which is harder to pattern
+  // match.
+  SmallVector<int64_t, 2> strideInts;
+  if (!matchPattern(op.stride(), m_TorchConstantIntList(strideInts)))
+    return rewriter.notifyMatchFailure(op, "only support constant int strides");
+  SmallVector<int64_t, 2> dilationInts;
+  if (!matchPattern(op.dilation(), m_TorchConstantIntList(dilationInts)))
+    return rewriter.notifyMatchFailure(op,
+                                       "only support constant int dilations");
+  SmallVector<int64_t, 2> paddingInts;
+  if (!matchPattern(op.padding(), m_TorchConstantIntList(paddingInts)))
+    return rewriter.notifyMatchFailure(op,
+                                       "only support constant int paddings");
+  SmallVector<int64_t, 2> kernelSizeInts;
+  if (!matchPattern(op.kernel_size(), m_TorchConstantIntList(kernelSizeInts)))
+    return rewriter.notifyMatchFailure(op, "only support kernel size ints");
+
+  Value falseValue = rewriter.create<arith::ConstantOp>(
+      loc, IntegerAttr::get(rewriter.getIntegerType(1), 0));
+  Value ceilModeFalse = rewriter.create<arith::CmpIOp>(
+      loc, arith::CmpIPredicate::eq, ceilMode, falseValue);
+  rewriter.create<AssertOp>(
+      loc, ceilModeFalse,
+      rewriter.getStringAttr("only ceil_mode false is supported"));
```

Comment:
Is there a reason for doing this at runtime? I think we can handle this at compile time by pattern matching on the boolean, similar to how we've done it in other ops

Reply:
ha, I think this was copied from the original code when refactoring. The thinking was runtime assert was more flexible and would work if it's not constant false but a runtime false. However, this is inconsistent with the rest of the codebase so it's better to be changed to the same as other places.

Reply:
Ya, I copied it from the original code. I will make the required changes.